h2integrate/storage/battery/pysam_battery.py

```diff
@@ -0,0 +1,495 @@
+            for _source_tech, intended_dispatch_tech in self.options["plant_config"][
+                "tech_to_dispatch_connections"
+            ]:
+                if any(intended_dispatch_tech in name for name in self.tech_group_name):
+                    self.add_discrete_input("pyomo_dispatch_solver", val=dummy_function)
+                    break
+
+        self.unmet_demand = 0.0
+        self.excess_resource = 0.0
+
+    def compute(self, inputs, outputs, discrete_inputs, discrete_outputs):
+        # Size the battery based on inputs -> method brought from HOPP
+        module_specs = {
+            "capacity": self.config.ref_module_capacity,
+            "surface_area": self.config.ref_module_surface_area,
+        }
+
+        self.size_batterystateful(
+            inputs["storage_capacity"][0],
+            self.system_model.ParamsPack.nominal_voltage,
+            module_specs=module_specs,
+        )
+        self.system_model.ParamsPack.h = 20
+        self.system_model.ParamsPack.Cp = 900
+        self.system_model.ParamsCell.resistance = 0.001
+        self.system_model.ParamsCell.C_rate = (
+            inputs["charge_rate"][0] / inputs["storage_capacity"][0]
+        )
+
+        # Minimum set of parameters to set to get statefulBattery to work
+        self._set_control_mode()
+
+        self.system_model.value("dt_hr", self.config.dt)
+        self.system_model.value("minimum_SOC", self.config.min_charge_percent * 100)
+        self.system_model.value("maximum_SOC", self.config.max_charge_percent * 100)
+        self.system_model.value("initial_SOC", self.config.init_charge_percent * 100)
+
+        # Setup PySAM battery model using PySAM method
+        self.system_model.setup()
+
+        # Run PySAM battery model 1 timestep to initialize values
+        self.system_model.value("dt_hr", 1.0)
+        self.system_model.value("input_power", 0.0)
+        self.system_model.execute(0)
+
+        if "pyomo_dispatch_solver" in discrete_inputs:
+            # Simulate the battery with provided dispatch inputs
+            dispatch = discrete_inputs["pyomo_dispatch_solver"]
+            kwargs = {
+                "time_step_duration": self.config.dt,
+                "control_variable": discrete_inputs["control_variable"],
+            }
+            dispatch(self.simulate, kwargs, inputs)
+        else:
+            # Simulate the battery with provided inputs
+            self.simulate(
+                electricity_in=inputs["electricity_in"],
+                demand_in=inputs["demand_in"],
+                time_step_duration=self.config.dt,
+                control_variable=discrete_inputs["control_variable"],
+            )
+        # Store outputs from the battery model
+        outputs["electricity_out"] = (
+            list(
+                np.array([P if P > 0.0 else 0.0 for P in self.outputs.P])
+                + np.array(self.outputs.excess_resource)
+            )
+        )
```

Comment:
Nitpick:
```suggestion
        # Store outputs from the battery model
        P = np.array(self.outputs.P)
        # Set negative power values to zero 
        P[P < 0] = 0.0
        outputs["electricity_out"] = P + np.array(self.outputs.excess_resource)
```

Double-check that it's the right logic, but I like the array usage and readability of this version

Reply:
@jaredthomas68

Reply:
Calculations like this have been moved to dispatch